pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -334,9 +311,64 @@ public DataType toDataType() {
       }
     }
 
+    /**
+     * Converts to the internal representation of the value.
```

Comment:
What is the meaning of internal and external in this context? It would be great to either specify that here or to include a link to some definition

Reply:
Good point. Added more comments